Missing error.h during the build

**Bug description**

Tarantool doesn't compile since the [commit](https://github.com/tarantool/tarantool/commit/4a12312c5723fcf1df64f993fefa1c0b16a15b1d) with the following error:
```
FAILED: [code=1] src/lib/core/CMakeFiles/core.dir/opt_def.c.o
ccache /usr/bin/cc -DCORO_ASM -DDT_NAMESPACE=tnt_ -DFIBER_STACK_SIZE_DEFAULT=528384 -DLUAJIT_ENABLE_GC64 -DLUAJIT_SMART_STRINGS=1 -DNDEBUG -DNVALGRIND -DNVALGRIND=1 -D__STDC_CONSTANT_MACROS=1 -D__STDC_FORMAT_MACROS=1 -D__STDC_LIMIT_MACROS=1 -I/Users/l.kats/tarantool_work2/build/build/curl/dest/include -I/Users/l.kats/tarantool_work2/src -I/Users/l.kats/tarantool_work2/build/src -I/Users/l.kats/tarantool_work2/src/lib -I/Users/l.kats/tarantool_work2/src/lib/small/include -I/Users/l.kats/tarantool_work2/build/src/lib/small/small/include -I/Users/l.kats/tarantool_work2/src/lib/small/third_party -I/Users/l.kats/tarantool_work2/src/lib/core -I/Users/l.kats/tarantool_work2/src/lib/digest -I/Users/l.kats/tarantool_work2/third_party -I/Users/l.kats/tarantool_work2/third_party/zstd/lib -I/Users/l.kats/tarantool_work2/third_party/zstd/lib/common -I/usr/local/opt/openssl@3/include -I/Users/l.kats/tarantool_work2/third_party/xxHash -I/Users/l.kats/tarantool_work2/third_party/luajit/src -I/Users/l.kats/tarantool_work2/build/third_party -I/Users/l.kats/tarantool_work2/third_party/coro -I/Users/l.kats/tarantool_work2/build/third_party/luajit/src -I/usr/local/opt/readline/include -I/Users/l.kats/tarantool_work2/build/build/libyaml/include -I/Users/l.kats/tarantool_work2/src/lib/msgpuck -I/usr/local/opt/icu4c@78/include -I/Users/l.kats/tarantool_work2/build/third_party/decNumber -I/Users/l.kats/tarantool_work2/third_party/libutil_freebsd -I/Users/l.kats/tarantool_work2/third_party/c-dt -F/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks -fexceptions -funwind-tables -fasynchronous-unwind-tables -fno-common -msse2 -Wformat -Wformat-security -Werror=format-security -fstack-protector-strong -fPIC -fmacro-prefix-map=/Users/l.kats/tarantool_work2=. -std=c11 -Wall -Wextra -Wno-gnu-alignof-expression -Wno-cast-function-type -O2 -g -DNDEBUG -ggdb -O2 -MD -MT src/lib/core/CMakeFiles/core.dir/opt_def.c.o -MF src/lib/core/CMakeFiles/core.dir/opt_def.c.o.d -o src/lib/core/CMakeFiles/core.dir/opt_def.c.o -c /Users/l.kats/tarantool_work2/src/lib/core/opt_def.c
/Users/l.kats/tarantool_work2/src/lib/core/opt_def.c:36:10: fatal error: 'error.h' file not found
   36 | #include "error.h"
      |          ^~~~~~~~~
1 error generated.
```

* OS: MacOS
* OS Version: 26.2
* Architecture: amd64

`tarantool --version` output from the **one commit before**.

```
Tarantool 3.7.0-entrypoint-56-g75c8e9e610
Target: Darwin-x86_64-RelWithDebInfo
Build options: cmake . -DCMAKE_INSTALL_PREFIX=/usr/local -DENABLE_BACKTRACE=TRUE
Compiler: AppleClang-17.0.0.17000603
C_FLAGS: -fexceptions -funwind-tables -fasynchronous-unwind-tables -fno-common -msse2 -Wformat -Wformat-security -Werror=format-security -fstack-protector-strong -fPIC -fmacro-prefix-map=/Users/l.kats/tarantool_work2=. -std=c11 -Wall -Wextra -Wno-gnu-alignof-expression -Wno-cast-function-type -O2 -g -DNDEBUG -ggdb -O2
CXX_FLAGS: -fexceptions -funwind-tables -fasynchronous-unwind-tables -fno-common -msse2 -Wformat -Wformat-security -Werror=format-security -fstack-protector-strong -fPIC -fmacro-prefix-map=/Users/l.kats/tarantool_work2=. -std=c++17 -Wall -Wextra -Wno-invalid-offsetof -Wno-gnu-alignof-expression -Wno-cast-function-type -O2 -g -DNDEBUG -ggdb -O2
```

**Steps to reproduce**

```sh
git checkout 4a12312c5723fcf1df64f993fefa1c0b16a15b1d
cmake .. -DCMAKE_BUILD_TYPE=RelWithDebInfo -GNinja -DLUAJIT_NO_UNWIND=ON
cmake --build .
```

--- a/src/lib/core/opt_def.c
+++ b/src/lib/core/opt_def.c
@@ -33,7 +33,6 @@
 #include "msgpuck.h"
 #include "bit/bit.h"
 #include "small/region.h"
-#include "error.h"
 #include "diag.h"
 #include "tt_static.h"
 

--- a/src/lib/core/tt_compression.c
+++ b/src/lib/core/tt_compression.c
@@ -6,7 +6,6 @@
 #include "tt_compression.h"
 #include "trivia/config.h"
 #include "diag.h"
-#include "error.h"
 #include "msgpuck.h"
 
 #if defined(ENABLE_TUPLE_COMPRESSION)

